ale_linters/markdown/mdl.vim

```diff
@@ -17,18 +17,17 @@ function! ale_linters#markdown#mdl#GetCommand(buffer) abort
     let l:options = ale#Var(a:buffer, 'markdown_mdl_options')
 
     return ale#Escape(l:executable) . l:exec_args
-    \   . (!empty(l:options) ? ' ' . l:options : '')
+    \   . ' -j' . (!empty(l:options) ? ' ' . l:options : '')
 endfunction
 
 function! ale_linters#markdown#mdl#Handle(buffer, lines) abort
-    " matches: '(stdin):173: MD004 Unordered list style'
-    let l:pattern = ':\(\d*\): \(.*\)$'
+    let l:errors = json_decode(join(a:lines, ''))
     let l:output = []
 
-    for l:match in ale#util#GetMatches(a:lines, l:pattern)
+    for l:error in l:errors
         call add(l:output, {
-        \   'lnum': l:match[1] + 0,
-        \   'text': l:match[2],
+        \   'lnum': l:error['line'],
+        \   'text': l:error['rule']  . '/' . join(l:error['aliases'], '/'). ' ' . l:error['description'],
```

Comment:
`rule` looks like an error code, so you should probably put it in `'code'`. See `:help ale-loclist-format`.